src/test/java/TasksServletTest.java

```diff
@@ -0,0 +1,146 @@
+import java.security.GeneralSecurityException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/**
+ * Test Tasks Servlet responds to client with correctly parsed Task names. Assumes
+ * AuthenticatedHttpServlet is functioning properly (those tests will fail otherwise).
+ */
+@RunWith(JUnit4.class)
+public final class TasksServletTest {
+  private AuthenticationVerifier authenticationVerifier;
+  private TasksClientFactory tasksClientFactory;
+  private TasksClient tasksClient;
+  private TasksServlet servlet;
+
+  private static final boolean AUTHENTICATION_VERIFIED = true;
+  private static final String ID_TOKEN_KEY = "idToken";
+  private static final String ID_TOKEN_VALUE = "sampleId";
+  private static final String ACCESS_TOKEN_KEY = "accessToken";
+  private static final String ACCESS_TOKEN_VALUE = "sampleAccessToken";
+  private static final Cookie sampleIdTokenCookie = new Cookie(ID_TOKEN_KEY, ID_TOKEN_VALUE);
+  private static final Cookie sampleAccessTokenCookie =
+      new Cookie(ACCESS_TOKEN_KEY, ACCESS_TOKEN_VALUE);
+  private static final Cookie[] validCookies =
+      new Cookie[] {sampleIdTokenCookie, sampleAccessTokenCookie};
+
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+
+  // Tasks must be returned in order of retrieval - JSON includes tasks in desired order
+  private static final String TASK_TITLE_ONE = "task one";
+  private static final String TASK_TITLE_TWO = "task two";
+  private static final String TASK_TITLE_THREE = "task three";
+  private static final String TASK_TITLE_FOUR = "task four";
+  private static final String TASK_ONE_TWO_JSON =
+      String.format("[{\"title\":\"%s\"},{\"title\":\"%s\"}]", TASK_TITLE_ONE, TASK_TITLE_TWO);
+  private static final String TASK_ALL_JSON =
+      String.format(
+          "[{\"title\":\"%s\"},{\"title\":\"%s\"},{\"title\":\"%s\"},{\"title\":\"%s\"}]",
+          TASK_TITLE_ONE, TASK_TITLE_TWO, TASK_TITLE_THREE, TASK_TITLE_FOUR);
+  private static final String EMPTY_JSON = "[]";
+  private static final List<TaskList> NO_TASK_LISTS = ImmutableList.of();
+  private static final List<TaskList> SOME_TASK_LISTS =
+      ImmutableList.of(new TaskList(), new TaskList());
+  private static final List<Task> NO_TASKS = ImmutableList.of();
+  private static final List<Task> TASKS_ONE_TWO =
+      ImmutableList.of(new Task().setTitle(TASK_TITLE_ONE), new Task().setTitle(TASK_TITLE_TWO));
+  private static final List<Task> TASKS_THREE_FOUR =
+      ImmutableList.of(new Task().setTitle(TASK_TITLE_THREE), new Task().setTitle(TASK_TITLE_FOUR));
```

Comment:
These specific task lists are only used in one test case each, and it would be more clear to inline the `ImmutableList.of` construction in the test cases.

Reply:
Agreed offline to keep as is